nn/conformer.py

```diff
@@ -0,0 +1,217 @@
+"""
+Conformer code.
+Ref: https://arxiv.org/abs/2005.08100
+"""
+
+from typing import Tuple, List, Union
+from .. import nn
+from . import LayerRef
+
+
+class _PositionwiseFeedForward(nn.Module):
+  """
+  Conformer position-wise feedforward neural network layer
+      FF -> Activation -> Dropout -> FF
+  """
+
+  def __init__(self, dim_model: int, dim_ff: int, dropout: float, activation, l2: float = 0.0):
+    """
+    :param dim_model:
+    :param dim_ff:
+    :param dropout:
+    :param activation:
+    :param l2:
+    """
+    super().__init__()
+
+    self.dropout = dropout
+    self.activation = activation
+
+    self.linear1 = nn.Linear(n_out=dim_ff, l2=l2)
+    self.linear2 = nn.Linear(n_out=dim_model, l2=l2)
+
+  def forward(self, inp: LayerRef) -> LayerRef:
+    return self.linear2(nn.dropout(self.activation(self.linear1(inp)), dropout=self.dropout))
+
+
+class _ConformerConvBlock(nn.Module):
+  """
+  Conformer convolution block
+      FF -> GLU -> depthwise conv -> BN -> Swish -> FF
+  """
+
+  def __init__(self, dim_model: int, kernel_size: int, l2: float = 0.0, batch_norm_eps: float = 1e-5,
+      batch_norm_momentum: float = 0.1, batch_norm_update_sample_only_in_training=True,
+      batch_norm_delay_sample_update=True, batch_norm_other_opts=None):
```

Comment:
Why do you have batch norm custom default options here? Why not just a single `batch_norm_other_opts`?

Reply:
so here first i want to remove `batch_norm_update_sample_only_in_training=True, batch_norm_delay_sample_update=True`. Because these should be True always otherwise batch norm logic is wrong. 
I thought it is nice to see the default params while calling this module as these two params will be the ones mainly tuned just in case.

Reply:
The wrong defaults should be fixed directly in the `BatchNorm` module, not here.

Reply:
Hm, I'm unsure if it is a good idea to have `batch_norm_momentum` and `batch_norm_eps` here...